Thomas-code/DFS_edges.ipynb

```diff
@@ -0,0 +1,1210 @@
+    "g.addEdge(12.7, [7.8])\n",
+    "g.addEdge(1.14, [14.15])\n",
+    "g.addEdge(14.15, [])\n",
+    "g.addEdge(21.22, [])\n",
+    "\"\"\"\n",
+    "        tr_nodes = [1]\n",
+    "    elif power_point_slide==29:\n",
+    "        graph_str = \"\"\"\n",
+    "g.addEdge(1.2, [2.3, 2.23])\n",
+    "g.addEdge(2.3, [3.7, 3.4])\n",
+    "g.addEdge(3.7, [7.8])\n",
+    "g.addEdge(7.8, [8.9])\n",
+    "g.addEdge(8.9, [9.5])\n",
+    "g.addEdge(9.5, [5.6])\n",
+    "g.addEdge(3.4, [4.5])\n",
+    "g.addEdge(4.5, [5.6])\n",
+    "g.addEdge(5.6, [])\n",
+    "g.addEdge(2.23, [23.11, 23.12])\n",
+    "g.addEdge(23.11, [11.12])\n",
+    "g.addEdge(11.12, [12.13, 12.7])\n",
+    "g.addEdge(23.12, [12.13, 12.7])\n",
+    "g.addEdge(12.13, [])\n",
+    "g.addEdge(1.14, [14.15])\n",
+    "g.addEdge(14.15, [])\n",
+    "g.addEdge(21.22, [])\n",
+    "g.addEdge(12.7, [7.8])\n",
+    "\"\"\"\n",
+    "        tr_nodes = [1,6]\n",
+    "    else:\n",
+    "        graph_str = 'does not exist'\n",
+    "        tr_nodes = [0]\n",
+    "        \n",
+    "    return graph_str, tr_nodes\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "b8d717f8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# build a graph with potentially random order of the entries\n",
+    "def build_graph(g, graph_str, shuffle = False):\n",
+    "    # create a list with possible graph entries\n",
+    "    graph_entries = []\n",
+    "    for i in graph_str.splitlines():\n",
+    "        if len(i)>0:\n",
+    "            # check if more than one edge can be reached from the starting edge;\n",
+    "            neighbor_str = i.split('[')[1].split(']')[0]\n",
+    "            neighbor_list = neighbor_str.split(',')\n",
+    "            nbr_neighbors = len(neighbor_list)\n",
+    "            if nbr_neighbors>1:\n",
+    "                for j in range(nbr_neighbors):\n",
+    "                    graph_entries.append(i.split('[')[0] + '[' + neighbor_list[j].replace(' ','') + '])')\n",
+    "            else:\n",
+    "                # delete all empty sapces\n",
+    "                graph_entries.append(i.replace(' ',''))\n",
+    "\n",
+    "    # randomly reorder the graph entries\n",
+    "    if shuffle == True:\n",
+    "        import random\n",
+    "        random.shuffle(graph_entries)\n",
+    "    \n",
+    "    # interpret the string in list to build the graph\n",
+    "    for i in graph_entries:\n",
+    "        exec(i)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "67b0a6e7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ---------------------------------------------------------------   \n",
+    "from termcolor import colored\n",
+    "import sys\n",
+    "\n",
+    "# ---------------------------------------------------------------   \n",
+    "# Helper functions:\n",
+    "class operations():\n",
+    "        \n",
+    "    def suffix(self, value):\n",
```

Comment:
Might be better to name the function as `get_suffix`, `get_prefix`. `suffix` or `prefix` would look like properties of an object.